File: Frontend/src/app/services/Admin.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { catchError, Observable, tap } from 'rxjs';
import { Admin } from '../models/admin';
import { Chambre } from '../models/chambre';
import { Evenement } from '../models/evenement';

@Injectable({
  providedIn: 'root'
})
export class AdminService {
  private apiUrl = 'http://localhost:8082/api/admins';  // Base URL
  private currentAdmin: any = null;

  constructor(private http: HttpClient) {}

  // Connexion
  login(login: string, motDePasse: string): Observable<any> {
    const params = new HttpParams()
      .set('login', login)
      .set('motDePasse', motDePasse);

    return this.http.post<any>(`${this.apiUrl}/login`, null, { params }).pipe(
      tap((admin: any) => {
        this.currentAdmin = admin;
      })
    );
  }

  // Vérifier si connecté
  isLoggedIn(): boolean {
    return this.currentAdmin !== null;
  }

  // Déconnexion
  logout() {
    this.currentAdmin = null;
  }

  // Récupérer l’admin connecté
  getAdmin() {
    return this.currentAdmin;
  }

  // ===== CLIENTS =====
  getAllClients(): Observable<any[]> {
    return this.http.get<any[]>(`${this.apiUrl}/clients`);
  }

  // ===== RÉSERVATIONS =====
  getAllReservations(): Observable<any[]> {
    return this.http.get<any[]>(`${this.apiUrl}/reservations`);
  }

  // ===== CHAMBRES =====
  getAllChambres(): Observable<any[]> {
    return this.http.get<any[]>(`${this.apiUrl}/chambres`);
  }

  createChambre(chambre: Chambre): Observable<any> {
    return this.http.post(`${this.apiUrl}/chambres`, chambre, {
      headers: {
        'Content-Type': 'application/json'
      }
    });
  }
  

  updateChambre(id: number, chambre: any): Observable<any> {
    return this.http.put<any>(`${this.apiUrl}/chambres/${id}`, chambre);
  }

  deleteChambre(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/chambres/${id}`);
  }

  // ===== OFFRES =====
  getAllOffres(): Observable<any[]> {
    return this.http.get<any[]>(`${this.apiUrl}/offres`);
  }

  createOffre(offre: any): Observable<any> {
    return this.http.post<any>(`${this.apiUrl}/offres`, offre, {
      headers: {
        'Content-Type': 'application/json'
      }
    });
  }

  updateOffre(id: number, offre: any): Observable<any> {
    return this.http.put<any>(`${this.apiUrl}/offres/${id}`, offre);
  }

  deleteOffre(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/offres/${id}`);
  }

  // ===== ÉVÉNEMENTS =====
  getAllEvenements(): Observable<any[]> {
    return this.http.get<any[]>(`${this.apiUrl}/evenements`);
  }

   createEvenement(evenement: Evenement): Observable<any> {
    return this.http.post<any>(`${this.apiUrl}/evenements`, evenement, {
      headers: {
        'Content-Type': 'application/json'
      }
    });
  }

  updateEvenement(id: number, evenement: any): Observable<any> {
    return this.http.put<any>(`${this.apiUrl}/evenements/${id}`, evenement);
  }

  deleteEvenement(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/evenements/${id}`);
  }
}
  

File: Frontend/Backend/projet_hotel/src/main/java/com/hotel/backend/services/AdminService.java

package com.hotel.backend.services;

import com.hotel.backend.entites.Admin;
import com.hotel.backend.entites.Chambre;
import com.hotel.backend.entites.Client;
import com.hotel.backend.entites.Evenement;
import com.hotel.backend.entites.Offre;
import com.hotel.backend.entites.Reservation;

import java.util.List;

import java.util.Optional;

public interface AdminService {

	Admin login(String login, String motDePasse);

    List<Admin> getAllAdmins();
    Optional<Admin> getAdminById(int id);
    Admin createAdmin(Admin admin);
    Admin updateAdmin(int id, Admin admin);
    void deleteAdmin(int id);

    List<Client> getAllClients();
    List<Reservation> getAllReservations();

    Chambre createChambre(Chambre chambre);
    Chambre updateChambre(int id, Chambre chambre);
    void deleteChambre(int id);
    List<Chambre> getAllChambres();

    Offre createOffre(Offre offre);
    Offre updateOffre(long id, Offre offre);
    void deleteOffre(long id);
    List<Offre> getAllOffres();

    Evenement createEvenement(Evenement evenement);
    Evenement updateEvenement(long id, Evenement evenement);
    void deleteEvenement(long id);
    List<Evenement> getAllEvenements();
}
